### PR TITLE
Allow adding layer through url in composer

### LINKED ISSUE #85: support creating maps/new with layer param
maps/new?layer=geonode:foo

think about how to handle remote services as well

### PR DESCRIPTION
**Pull request by bartvde**
## What does this PR do?
Allow adding a layer to the composer through a url param, for example http://localhost:8000/maps/new?layer=geonode:lakes

## Screenshot
selection_046

## Related Issue
closes #85

### CODE REVIEW

**Comment by bartvde** [comment]
Using this approach, not all layer metadata is retrieved, such as the stylename etc.
Maybe we should do this automatically in sdk whenever a layer gets added to the store and it doesn't have this info. Currently this is spread in different parts of the codebase instead (AddLayerModal, MapConfig*Service).

**Comment by bartvde** [comment]
See: https://github.com/boundlessgeo/sdk/pull/363 for the sdk part

**Comment by bartvde** [comment]
Maybe we should run through getCapabilities, otherwise we cannot zoom to the layer

File: src/components/geonode.jsx [hunk 1]
```diff
@@ -63,10 +64,23 @@ class GeoNodeViewer extends React.Component {
...
     this.mode = this.props.mode || 'viewer';
     this.edit = (this.mode === 'composer');
+    if (this.props.layer) {
+      WMSService.getCapabilities(this.props.wmsServer, this._getLayerInfo.bind(this), function() {}, this.props.proxy);
+    }
   }
   componentWillReceiveProps(props) {
     this.updateMap(props);
   }
+  _getLayerInfo(layerInfo, getMapUrl) {
```
Comment (milafrerichs) [inline src/components/geonode.jsx]: should we move that into a service maybe? 
or into the SDK? and call the SDK to get the info for a specific layer?

Comment (bartvde) [reply src/components/geonode.jsx]: yes I thought about this as well, I'll move it into SDK

Comment (bartvde) [reply src/components/geonode.jsx]: Moved most of the code to sdk with: https://github.com/boundlessgeo/sdk/pull/364

**Comment by bartvde** [comment]
I'll leave this open until we have a new SDK release that includes the new function

**Review by milafrerichs — APPROVED** [review]

**Comment by bartvde** [comment]
Going to merge this, SDK upgrade is coming today.

**Pull request merged**